test/banach/retry_test.clj

```diff
@@ -5,6 +5,40 @@
             [clojure.math.numeric-tower :as math]
             [banach.retry :as retry]))
 
+(def ctx-1 {:failures [(Exception. "Slow your roll")]})
+(def ctx-2 (update ctx-1 :failures conj (Exception. "KABOOM")))
+
+(deftest routing-tests
+  (let [strat (retry/routing
+               (comp odd? count :failures) identity
+               (constantly true) retry/give-up)]
```

Comment:
I think it may have been the final pair `(constantly true) retry/give-up`. It makes sense when I look at it now, but it was a bit confusing to think that we need to have a bit of special handling to effectively attach an errback at the end of the strategy.